src/Client/Grpc/GrpcDurableEntityClient.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Google.Protobuf.WellKnownTypes;
+using Microsoft.DurableTask.Client.Entities;
+using Microsoft.DurableTask.Entities;
+using Microsoft.Extensions.Logging;
+using static Microsoft.DurableTask.Protobuf.TaskHubSidecarService;
+using DTCore = DurableTask.Core;
+using P = Microsoft.DurableTask.Protobuf;
+
+namespace Microsoft.DurableTask.Client.Grpc;
+
+/// <summary>
+/// The client for entities.
+/// </summary>
+class GrpcDurableEntityClient : DurableEntityClient
+{
+    readonly TaskHubSidecarServiceClient sidecarClient;
+    readonly GrpcDurableTaskClient durableTaskClient;
+    readonly ILogger logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GrpcDurableEntityClient"/> class.
+    /// </summary>
+    /// <param name="durableTaskClient">The durable Task client.</param>
+    /// <param name="sidecarClient">The client for the GRPC connection to the sidecar.</param>
+    /// <param name="logger">The logger for logging client requests.</param>
+    public GrpcDurableEntityClient(GrpcDurableTaskClient durableTaskClient, TaskHubSidecarServiceClient sidecarClient, ILogger logger)
+        : base(durableTaskClient.Name)
+    {
+        this.durableTaskClient = durableTaskClient;
+        this.sidecarClient = sidecarClient;
+        this.logger = logger;
+    }
+
+    /// <inheritdoc/>
+    public override async Task SignalEntityAsync(EntityInstanceId id, string operationName, object? input = null, SignalEntityOptions? options = null, CancellationToken cancellation = default)
+    {
+        Guid requestId = Guid.NewGuid();
+        DateTimeOffset? scheduledTime = options?.SignalTime;
+
+        P.SignalEntityRequest request = new P.SignalEntityRequest()
+        {
+            InstanceId = id.ToString(),
+            RequestId = Google.Protobuf.ByteString.FromStream(new MemoryStream(requestId.ToByteArray())),
+            Name = operationName,
+            Input = this.durableTaskClient.DataConverter.Serialize(input),
+            ScheduledTime = scheduledTime.HasValue ? Timestamp.FromDateTimeOffset(scheduledTime.Value.ToUniversalTime()) : default,
```

Comment:
I think we have an extension method for converting `DateTimeOffset?` to `Timestamp`
https://github.com/microsoft/durabletask-dotnet/blob/main/src/Shared/Grpc/ProtoUtils.cs#L216

```suggestion
            ScheduledTime = scheduledTime.ToTimestamp(),
```

Reply:
Since it converts to `ScheduledTime?`, not `ScheduledTime`, I have to use `scheduledTime.ToTimestamp() ?? default`. Still looks nicer.